src/services/lineaStateBridgeEventListener.js

```diff
@@ -0,0 +1,76 @@
+import { ethers } from "ethers";
+
+export const lineaStateBridgeEventListener = async ({
+  ethereumEndpoint,
+  lineaStateBridgeAddress,
+  l1MessageServiceAddress,
+}) => {
+  let provider;
+  let reconnectAttempts = 0;
+  const MAX_RECONNECT_ATTEMPTS = 10;
+  const RECONNECT_DELAY = 5000; // 5 seconds delay between reconnection attempts
+
+  const initializeProvider = () => {
+    provider = new ethers.providers.JsonRpcProvider(ethereumEndpoint);
+
+    // Listen for provider events
+    provider.on('connect', () => {
+      console.log('Connected to the Ethereum network.');
+      reconnectAttempts = 0; // Reset attempts on successful connection
+    });
+
+    provider.on('error', (error) => {
+      console.error('Network error:', error);
+    });
+
+    provider.on('disconnect', (error) => {
+      console.error('Disconnected from the Ethereum network:', error);
+      handleReconnection();
+    });
+  };
+
+  // Function to handle reconnection
+  const handleReconnection = () => {
+    if (reconnectAttempts < MAX_RECONNECT_ATTEMPTS) {
+      reconnectAttempts++;
+      console.log(`Attempting to reconnect... (${reconnectAttempts}/${MAX_RECONNECT_ATTEMPTS})`);
+      setTimeout(() => {
+        initializeProvider();
+        listenForEvents(); // Reattach event listeners after reconnection
+      }, RECONNECT_DELAY);
+    } else {
+      console.error('Max reconnection attempts reached. Exiting...');
+      process.exit(1); // Exit process after max attempts
+    }
+  };
+
+  // Function to listen for events
+  const listenForEvents = () => {
+    // Event ABI
+    const abi = ["event MessageSent (uint256 _nonce, bytes _calldata, index_topic_3 bytes32 _messageHash)"];
```

Comment:
This is an incomplete ABI for the MessageSent event. The complete one looks like this:
```suggestion
    const abi = [
    "event MessageSent(address indexed _from, address indexed _to, uint256 _fee, uint256 _value, uint256 _nonce, bytes _calldata, bytes32 indexed _messageHash)"
  ];
```
It can be found here: https://github.com/Consensys/linea-contracts/blob/main/contracts/interfaces/IMessageService.sol